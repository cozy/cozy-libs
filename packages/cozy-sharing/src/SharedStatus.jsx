import React from 'react'

import { AvatarList } from './components/Avatar/AvatarList'
import SharingContext from './context'
import withLocales from './hoc/withLocales'

export const SharedStatus = withLocales(
  ({ docId, className, noSharedClassName, onClick, showMeAsOwner }) => (
    <SharingContext.Consumer>
      {({ byDocId, getRecipients, getSharingLink, isOwner } = {}) =>
        !byDocId || !byDocId[docId] ? (
          <span className={className + ' ' + noSharedClassName}>â€”</span>
        ) : (
          <AvatarList
            className={className}
            recipients={getRecipients(docId)}
            link={getSharingLink(docId) !== null}
            onClick={onClick}
            isOwner={isOwner(docId)}
            size="s"
            showMeAsOwner={showMeAsOwner}
          />
        )
      }
    </SharingContext.Consumer>
  )
)
