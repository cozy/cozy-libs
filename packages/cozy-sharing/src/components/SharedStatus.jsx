import React from 'react'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import palette from 'cozy-ui/transpiled/react/palette'
import { SharingTooltip, TooltipRecipientList } from './Tooltip'
import cx from 'classnames'

import { getDisplayName } from '../models'
import styles from './status.styl'
import LinkIcon from '../../assets/icons/icon-link.svg'

export const SharedStatus = ({ className, docId, recipients, link }) => {
  const { t } = useI18n()
  return (
    <span className={cx(className, styles['shared-status'])}>
      {recipients.length > 1 && (
        <a
          data-tip
          data-for={`members${docId}`}
          className={styles['shared-status-label']}
        >
          {t('Share.members.count', { smart_count: recipients.length })}
        </a>
      )}
      {recipients.length > 1 && (
        <SharingTooltip id={`members${docId}`}>
          <TooltipRecipientList
            recipientNames={recipients.map(recipient =>
              getDisplayName(recipient)
            )}
          />
        </SharingTooltip>
      )}
      {link && (
        <>
          <LinkIcon
            style={{ fill: palette.coolGrey }}
            data-tip
            data-for={`linkfor${docId}`}
          />

          <SharingTooltip id={`linkfor${docId}`}>
            <span>{t('Files.share.sharedByLink')}</span>
          </SharingTooltip>
        </>
      )}
    </span>
  )
}

export default SharedStatus
