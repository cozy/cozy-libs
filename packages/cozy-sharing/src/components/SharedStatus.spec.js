import React from 'react'
import { mount } from 'enzyme'
import { SharedStatus } from './SharedStatus'
import AppLike from '../../test/AppLike'
describe('SharedStatus component', () => {
  const WrappingComponent = ({ children }) => <AppLike>{children}</AppLike>

  it('should just render a span if no sharing', () => {
    const component = mount(<SharedStatus docId="1" recipients={[]} />, {
      wrappingComponent: WrappingComponent
    })
    expect(component).toMatchSnapshot()
  })

  it('should have the right display if there is several recipients', () => {
    const component = mount(
      <SharedStatus
        docId="1"
        recipients={[
          {
            _id: 1,
            name: '1'
          },
          {
            _id: 2,
            name: '2'
          }
        ]}
      />,
      {
        wrappingComponent: WrappingComponent
      }
    )
    expect(component).toMatchSnapshot()
  })

  it('should display the link if there is a link', () => {
    const component = mount(
      <SharedStatus docId="1" recipients={[]} link={true} />,
      {
        wrappingComponent: WrappingComponent
      }
    )
    expect(component).toMatchSnapshot()
  })
})
