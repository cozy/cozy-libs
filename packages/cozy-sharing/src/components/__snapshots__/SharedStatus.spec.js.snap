// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedStatus component should display the link if there is a link 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <SharedStatus
        docId="1"
        link={true}
        recipients={Array []}
        t={[Function]}
      >
        <span
          className="shared-status"
        >
          <LinkIcon
            data-for="linkfor1"
            data-tip={true}
            height="16"
            style={
              Object {
                "fill": "var(--coolGrey)",
              }
            }
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <svg
              data-for="linkfor1"
              data-tip={true}
              height="16"
              style={
                Object {
                  "fill": "var(--coolGrey)",
                }
              }
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <defs>
                <path
                  d="M8.88 3.88a1 1 0 1 1-1.415-1.415L9.053.877A2.998 2.998 0 0 1 13.29.876l1.18 1.18a2.998 2.998 0 0 1-.007 4.24L10.999 9.76c-1.535 1.535-4.07 1.583-5.534.12A1 1 0 1 1 6.88 8.465c.67.67 1.94.646 2.705-.119l3.465-3.464a.998.998 0 0 0 .006-1.413l-1.18-1.18a.998.998 0 0 0-1.408.003L8.88 3.88zm-2.415 7.585A1 1 0 1 1 7.88 12.88l-1.587 1.587a2.998 2.998 0 0 1-4.237.002l-1.18-1.18a2.998 2.998 0 0 1 .007-4.24l3.464-3.464c1.534-1.535 4.07-1.583 5.533-.12A1 1 0 1 1 8.465 6.88c-.67-.67-1.94-.646-2.705.119l-3.464 3.464a.998.998 0 0 0-.006 1.413l1.18 1.18a.998.998 0 0 0 1.407-.003l1.588-1.588z"
                  id="link-1-path-1"
                />
              </defs>
              <use
                fillRule="nonzero"
                xlinkHref="#link-1-path-1"
              />
            </svg>
          </LinkIcon>
          <SharingTooltip
            id="linkfor1"
          >
            <ReactTooltip
              className="shared-tooltip"
              clickable={false}
              effect="solid"
              id="linkfor1"
              insecure={true}
              place="bottom"
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-bottom type-dark "
                data-id="tooltip"
                id="linkfor1"
              >
                <span>
                  Files.share.shareByLink.subtitle
                </span>
              </div>
            </ReactTooltip>
          </SharingTooltip>
        </span>
      </SharedStatus>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;

exports[`SharedStatus component should have the right display if there is several recipients 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <SharedStatus
        docId="1"
        recipients={
          Array [
            Object {
              "_id": 1,
              "name": "1",
            },
            Object {
              "_id": 2,
              "name": "2",
            },
          ]
        }
        t={[Function]}
      >
        <span
          className="shared-status"
        >
          <a
            className="shared-status-label"
            data-for="members1"
            data-tip={true}
          >
            Share.members.count
          </a>
          <SharingTooltip
            id="members1"
          >
            <ReactTooltip
              className="shared-tooltip"
              clickable={false}
              effect="solid"
              id="members1"
              insecure={true}
              place="bottom"
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-bottom type-dark "
                data-id="tooltip"
                id="members1"
              >
                <TooltipRecipientList
                  cutoff={4}
                  recipientNames={
                    Array [
                      "1",
                      "2",
                    ]
                  }
                >
                  <ul
                    className="shared-tooltip-list"
                  >
                    <li
                      key="key_name_1_0"
                    >
                      1
                    </li>
                    <li
                      key="key_name_2_1"
                    >
                      2
                    </li>
                  </ul>
                </TooltipRecipientList>
              </div>
            </ReactTooltip>
          </SharingTooltip>
        </span>
      </SharedStatus>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;

exports[`SharedStatus component should just render a span if no sharing 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <SharedStatus
        docId="1"
        recipients={Array []}
      >
        <span
          className="shared-status"
        />
      </SharedStatus>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;
